File: rearend/iOSClub.WebAPI/iOSClub.Data/DataModels/ArticleModel.cs
using System.ComponentModel.DataAnnotations;

namespace iOSClub.Data.DataModels;

public class ArticleModel
{
    /// <summary>
    /// 路径
    /// </summary>
    [Key]
    [MaxLength(128)]
    public string Path { get; set; } = "";

    /// <summary>
    /// 标题
    /// </summary>
    [MaxLength(32)]
    public string Title { get; set; } = "";

    /// <summary>
    /// 内容
    /// </summary>
    public string Content { get; set; } = "";

    /// <summary>
    /// 最后修改时间
    /// </summary>
    public DateTime LastWriteTime { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// Founder : 创始人
    /// President : 社长,团支书
    /// Minister : 部长
    /// Department : 部员成员
    /// </summary>
    [MaxLength(20)]
    public string? Identity { get; set; }

    /// <summary>
    /// 分类ID
    /// </summary>
    [MaxLength(128)]
    public string? CategoryId { get; set; }

    /// <summary>
    /// 分类
    /// </summary>
    public CategoryModel? Category { get; set; }

    /// <summary>
    /// 文章排序（用于自定义分类内文章顺序）
    /// </summary>
    public int ArticleOrder { get; set; }
}

File: frontend/src/models/ArticleModel.ts
// 分类模型接口
export interface CategoryModel {
  id: string;
  name: string;
  order: number;
  description?: string;
}

// 文章模型接口 —— 对应后端 VOs/ArticleVO.cs（列表为 ArticleListItemVO.cs）
// 后端返回的是扁平的 categoryId / categoryName，没有嵌套的 category 对象。
export interface ArticleModel {
  path: string;
  title: string;
  content: string;
  lastWriteTime: string;
  identity?: string;
  visibleToDepartment?: string;
  categoryId?: string;
  categoryName?: string;
  articleOrder?: number;
}

// 文章创建请求 —— 对应后端 DTOs/ArticleCreateDTO.cs
// 注意 category 传的是分类**名称**，后端没有 categoryId 字段。
export interface ArticleCreateDto {
  path: string;
  title: string;
  content: string;
  identity?: string;
  visibleToDepartment?: string;
  articleOrder?: number;
  category?: string;
}

// 文章更新请求 —— 对应后端 DTOs/ArticleUpdateDTO.cs（path 走路由，不在 body 里）
export interface ArticleUpdateDto{
  title: string;
  content: string;
  identity?: string;
  visibleToDepartment?: string;
  articleOrder?: number;
  category?: string;
}

// 文章搜索结果接口 —— 对应后端 ArticleRepository.ArticleSearchResult
export interface ArticleSearchResult {
  path: string;
  title: string;
  content: string;
  lastWriteTime: string;
  identity?: string;
  visibleToDepartment?: string;
  categoryId?: string;
  categoryName?: string;
  articleOrder?: number;
  highlight?: string;
  highlightedTitle?: string;
  highlightedContent?: string;
}
